docs/user_guide/issues.md

```diff
@@ -1 +1,117 @@
 # Understanding issues
+
+BIDS Validator issues are structured results that describe problems or potential
+problems in the dataset.
+Issues can either be warnings or errors.
+
+Errors are emitted when a violation of the BIDS specification is detected,
+or, in some cases, the validator is unable to read part of the dataset.
+Errors must be resolved in order for a dataset to be considered valid.
+
+Warnings are emitted for violations of BIDS specification recommendations,
+as well as cases where a common problem may be present, and user attention
+is needed to verify the validity of the dataset.
+
+## Issue model
+
+Every issue has a code indicating the issue type, a location indicating
+the file where the issue was detected, and a severity field that can
+either be "warning" or "error".
```

Comment:
could it also be "ignore" or alike (I think @candleindark noted those as the ones which are told to be ignored by config, right?)  . Might be worth noting explicitly if so, e.g. 

```suggestion
either be "warning" or "error".
If some issues are configured to be ignored, in JSON output mode they would still be listed and
have severity "ignore".
```